Convert JS -> TS support

StrapiShaun opened the pull request:
<!--
Hello 👋 Thank you for submitting a pull request.

To help us merge your PR, make sure to follow the instructions below:

- Create or update the documentation. (Should be made against the `main` branch)
- Create or update the tests.
- Refer to the issue you are closing in the PR description - fix #issue
- Specify if the PR is in WIP (work in progress) state or ready to be merged

Please ensure you read through the Contributing Guide: 
https://github.com/strapi/documentation/blob/main/CONTRIBUTING.md
-->

### What does it do?

Adds a section to TypeScript docs for adding TS support to an existing project 

### Why is it needed?

Allow devs to migrate existing JS projects to TS-supported projects. 

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

Bassel17 reviewed (approved):
Looks good

pwizla commented on docs/developer-docs/latest/development/typescript.md:
```diff
@@ -6,14 +6,10 @@ canonicalUrl: https://docs.strapi.io/developer-docs/latest/development/typescrip
 # TypeScript development
 
-TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.2.0 and above. TypeScript-enabled projects allow developing plugins with TypeScript as well as using TypeScript typings.
+TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.3.0 and above. Existing JavaScript projects can [add TypeScript support](#add-typescript-support-to-an-existing-strapi-project) through a short migration procedure. TypeScript-enabled projects allow developing plugins ...
```
I'm wondering about the use of "short" 🤔 I tend to think we should avoid it, just like we avoid "simple" or "easy". Especially because if at some point a user has an issue, it could take a longer time, and they might be frustrated. 
Also, as we've just discussed it during the DX sync meeting, maybe "conversion" is better than "migration"? :)

pwizla commented on docs/developer-docs/latest/development/typescript.md:
```diff
@@ -132,3 +128,84 @@ const appContext = await strapi.compile();
+
+## Add TypeScript support to an existing Strapi project
+
+Adding TypeScript support to an existing project requires modifications to the project files. Additionally the TypeScript flag `allowJs` should be set to `true` in the root `tsconfig.json` file to incrementally add TypeScript files to existing JavaScript projects. The `allowJs` flag allows `.ts` and `.tsx` files to coexist with JavaScript files. TypeScript support can be added to an existing Stra ...
```
> Adding TypeScript support to an existing project requires modifications to the project files. 

I guess you mean that devs will need to (gradually) convert each JS file to a TS file and take advantage of TS features and syntax, right? Maybe we could be a bit more explicit, then, what do you think?

And this is also why I think the "conversion" to TS is not so "short" 😅 (see my other comment above). Adding the `tsconfig` files is fast, but converting all the files… not so much 😅 So I guess it really depends on the wording we chose.

StrapiShaun replied:
I see what you mean. I added a bit more detail to make it clear.

pwizla commented on docs/developer-docs/latest/development/typescript.md:
```diff
@@ -132,3 +128,84 @@ const appContext = await strapi.compile();
+
+## Add TypeScript support to an existing Strapi project
+
+Adding TypeScript support to an existing project requires modifications to the project files. Additionally the TypeScript flag `allowJs` should be set to `true` in the root `tsconfig.json` file to incrementally add TypeScript files to existing JavaScript projects. The `allowJs` flag allows `.ts` and `.tsx` files to coexist with JavaScript files. TypeScript support can be added to an existing Stra ...
```
```suggestion
Adding TypeScript support to an existing project requires modifications to the project files. Additionally the TypeScript flag `allowJs` should be set to `true` in the root `tsconfig.json` file to incrementally add TypeScript files to existing JavaScript projects. The `allowJs` flag allows `.ts` and `.tsx` files to coexist with JavaScript files.

TypeScript support can be added to an existing Strapi project using the following procedure:
```

This is a bit more of a personal preference, but in other documentation sections, the introduction/concept paragraph is clearly separated from the "introductory" sentence for the procedure. 🤓

pwizla reviewed (commented):
Thank you, Shaun. This will be really helpful to users!
I have a few questions, would be happy to read your opinion. 🤓

StrapiShaun commented:
> Thank you, Shaun. This will be really helpful to users! I have a few questions, would be happy to read your opinion. 🤓

Thanks @pwizla. I think I have addressed all of your comments. I cleaned up a couple of additional items as well.😀

pwizla commented:
LGTM, thanks for the updates, Shaun :)
